stan/math/rev/fun/value_of.hpp

```diff
@@ -20,7 +20,16 @@ namespace math {
  * @param v Variable.
  * @return Value of variable.
  */
-inline double value_of(const var& v) { return v.vi_->val_; }
+template <typename Var, require_var_t<Var>* = nullptr>
```

Comment:
We don't need requires here. Previous implementation was fine and in my opinion easier to read.